// @aws-sdk/client-cognito-identity-provider/dist-cjs/models/models_1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerifyUserAttributeResponse = exports.VerifyUserAttributeRequest = exports.VerifySoftwareTokenResponse = exports.VerifySoftwareTokenResponseType = exports.VerifySoftwareTokenRequest = exports.EnableSoftwareTokenMFAException = exports.UpdateUserPoolDomainResponse = exports.UpdateUserPoolDomainRequest = exports.UpdateUserPoolClientResponse = exports.UpdateUserPoolClientRequest = exports.UpdateUserPoolResponse = exports.UpdateUserPoolRequest = exports.UpdateUserAttributesResponse = exports.UpdateUserAttributesRequest = exports.UpdateResourceServerResponse = exports.UpdateResourceServerRequest = exports.UpdateIdentityProviderResponse = exports.UpdateIdentityProviderRequest = exports.UpdateGroupResponse = exports.UpdateGroupRequest = exports.UpdateDeviceStatusResponse = exports.UpdateDeviceStatusRequest = exports.UpdateAuthEventFeedbackResponse = exports.UpdateAuthEventFeedbackRequest = exports.UntagResourceResponse = exports.UntagResourceRequest = exports.TagResourceResponse = exports.TagResourceRequest = exports.StopUserImportJobResponse = exports.StopUserImportJobRequest = exports.StartUserImportJobResponse = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const CognitoIdentityProviderServiceException_1 = require("./CognitoIdentityProviderServiceException");
const models_0_1 = require("./models_0");
var StartUserImportJobResponse;
(function (StartUserImportJobResponse) {
    StartUserImportJobResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StartUserImportJobResponse = exports.StartUserImportJobResponse || (exports.StartUserImportJobResponse = {}));
var StopUserImportJobRequest;
(function (StopUserImportJobRequest) {
    StopUserImportJobRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StopUserImportJobRequest = exports.StopUserImportJobRequest || (exports.StopUserImportJobRequest = {}));
var StopUserImportJobResponse;
(function (StopUserImportJobResponse) {
    StopUserImportJobResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StopUserImportJobResponse = exports.StopUserImportJobResponse || (exports.StopUserImportJobResponse = {}));
var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagResourceRequest = exports.TagResourceRequest || (exports.TagResourceRequest = {}));
var TagResourceResponse;
(function (TagResourceResponse) {
    TagResourceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagResourceResponse = exports.TagResourceResponse || (exports.TagResourceResponse = {}));
var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagResourceRequest = exports.UntagResourceRequest || (exports.UntagResourceRequest = {}));
var UntagResourceResponse;
(function (UntagResourceResponse) {
    UntagResourceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagResourceResponse = exports.UntagResourceResponse || (exports.UntagResourceResponse = {}));
var UpdateAuthEventFeedbackRequest;
(function (UpdateAuthEventFeedbackRequest) {
    UpdateAuthEventFeedbackRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Username && { Username: smithy_client_1.SENSITIVE_STRING }),
        ...(obj.FeedbackToken && { FeedbackToken: smithy_client_1.SENSITIVE_STRING }),
    });
})(UpdateAuthEventFeedbackRequest = exports.UpdateAuthEventFeedbackRequest || (exports.UpdateAuthEventFeedbackRequest = {}));
var UpdateAuthEventFeedbackResponse;
(function (UpdateAuthEventFeedbackResponse) {
    UpdateAuthEventFeedbackResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateAuthEventFeedbackResponse = exports.UpdateAuthEventFeedbackResponse || (exports.UpdateAuthEventFeedbackResponse = {}));
var UpdateDeviceStatusRequest;
(function (UpdateDeviceStatusRequest) {
    UpdateDeviceStatusRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.AccessToken && { AccessToken: smithy_client_1.SENSITIVE_STRING }),
    });
})(UpdateDeviceStatusRequest = exports.UpdateDeviceStatusRequest || (exports.UpdateDeviceStatusRequest = {}));
var UpdateDeviceStatusResponse;
(function (UpdateDeviceStatusResponse) {
    UpdateDeviceStatusResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDeviceStatusResponse = exports.UpdateDeviceStatusResponse || (exports.UpdateDeviceStatusResponse = {}));
var UpdateGroupRequest;
(function (UpdateGroupRequest) {
    UpdateGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateGroupRequest = exports.UpdateGroupRequest || (exports.UpdateGroupRequest = {}));
var UpdateGroupResponse;
(function (UpdateGroupResponse) {
    UpdateGroupResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateGroupResponse = exports.UpdateGroupResponse || (exports.UpdateGroupResponse = {}));
var UpdateIdentityProviderRequest;
(function (UpdateIdentityProviderRequest) {
    UpdateIdentityProviderRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateIdentityProviderRequest = exports.UpdateIdentityProviderRequest || (exports.UpdateIdentityProviderRequest = {}));
var UpdateIdentityProviderResponse;
(function (UpdateIdentityProviderResponse) {
    UpdateIdentityProviderResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateIdentityProviderResponse = exports.UpdateIdentityProviderResponse || (exports.UpdateIdentityProviderResponse = {}));
var UpdateResourceServerRequest;
(function (UpdateResourceServerRequest) {
    UpdateResourceServerRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateResourceServerRequest = exports.UpdateResourceServerRequest || (exports.UpdateResourceServerRequest = {}));
var UpdateResourceServerResponse;
(function (UpdateResourceServerResponse) {
    UpdateResourceServerResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateResourceServerResponse = exports.UpdateResourceServerResponse || (exports.UpdateResourceServerResponse = {}));
var UpdateUserAttributesRequest;
(function (UpdateUserAttributesRequest) {
    UpdateUserAttributesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.UserAttributes && {
            UserAttributes: obj.UserAttributes.map((item) => models_0_1.AttributeType.filterSensitiveLog(item)),
        }),
        ...(obj.AccessToken && { AccessToken: smithy_client_1.SENSITIVE_STRING }),
    });
})(UpdateUserAttributesRequest = exports.UpdateUserAttributesRequest || (exports.UpdateUserAttributesRequest = {}));
var UpdateUserAttributesResponse;
(function (UpdateUserAttributesResponse) {
    UpdateUserAttributesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserAttributesResponse = exports.UpdateUserAttributesResponse || (exports.UpdateUserAttributesResponse = {}));
var UpdateUserPoolRequest;
(function (UpdateUserPoolRequest) {
    UpdateUserPoolRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserPoolRequest = exports.UpdateUserPoolRequest || (exports.UpdateUserPoolRequest = {}));
var UpdateUserPoolResponse;
(function (UpdateUserPoolResponse) {
    UpdateUserPoolResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserPoolResponse = exports.UpdateUserPoolResponse || (exports.UpdateUserPoolResponse = {}));
var UpdateUserPoolClientRequest;
(function (UpdateUserPoolClientRequest) {
    UpdateUserPoolClientRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.ClientId && { ClientId: smithy_client_1.SENSITIVE_STRING }),
    });
})(UpdateUserPoolClientRequest = exports.UpdateUserPoolClientRequest || (exports.UpdateUserPoolClientRequest = {}));
var UpdateUserPoolClientResponse;
(function (UpdateUserPoolClientResponse) {
    UpdateUserPoolClientResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.UserPoolClient && { UserPoolClient: models_0_1.UserPoolClientType.filterSensitiveLog(obj.UserPoolClient) }),
    });
})(UpdateUserPoolClientResponse = exports.UpdateUserPoolClientResponse || (exports.UpdateUserPoolClientResponse = {}));
var UpdateUserPoolDomainRequest;
(function (UpdateUserPoolDomainRequest) {
    UpdateUserPoolDomainRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserPoolDomainRequest = exports.UpdateUserPoolDomainRequest || (exports.UpdateUserPoolDomainRequest = {}));
var UpdateUserPoolDomainResponse;
(function (UpdateUserPoolDomainResponse) {
    UpdateUserPoolDomainResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserPoolDomainResponse = exports.UpdateUserPoolDomainResponse || (exports.UpdateUserPoolDomainResponse = {}));
class EnableSoftwareTokenMFAException extends CognitoIdentityProviderServiceException_1.CognitoIdentityProviderServiceException {
    constructor(opts) {
        super({
            name: "EnableSoftwareTokenMFAException",
            $fault: "client",
            ...opts,
        });
        this.name = "EnableSoftwareTokenMFAException";
        this.$fault = "client";
        Object.setPrototypeOf(this, EnableSoftwareTokenMFAException.prototype);
    }
}
exports.EnableSoftwareTokenMFAException = EnableSoftwareTokenMFAException;
var VerifySoftwareTokenRequest;
(function (VerifySoftwareTokenRequest) {
    VerifySoftwareTokenRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.AccessToken && { AccessToken: smithy_client_1.SENSITIVE_STRING }),
    });
})(VerifySoftwareTokenRequest = exports.VerifySoftwareTokenRequest || (exports.VerifySoftwareTokenRequest = {}));
var VerifySoftwareTokenResponseType;
(function (VerifySoftwareTokenResponseType) {
    VerifySoftwareTokenResponseType["ERROR"] = "ERROR";
    VerifySoftwareTokenResponseType["SUCCESS"] = "SUCCESS";
})(VerifySoftwareTokenResponseType = exports.VerifySoftwareTokenResponseType || (exports.VerifySoftwareTokenResponseType = {}));
var VerifySoftwareTokenResponse;
(function (VerifySoftwareTokenResponse) {
    VerifySoftwareTokenResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(VerifySoftwareTokenResponse = exports.VerifySoftwareTokenResponse || (exports.VerifySoftwareTokenResponse = {}));
var VerifyUserAttributeRequest;
(function (VerifyUserAttributeRequest) {
    VerifyUserAttributeRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.AccessToken && { AccessToken: smithy_client_1.SENSITIVE_STRING }),
    });
})(VerifyUserAttributeRequest = exports.VerifyUserAttributeRequest || (exports.VerifyUserAttributeRequest = {}));
var VerifyUserAttributeResponse;
(function (VerifyUserAttributeResponse) {
    VerifyUserAttributeResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(VerifyUserAttributeResponse = exports.VerifyUserAttributeResponse || (exports.VerifyUserAttributeResponse = {}));
