// @aws-sdk/client-cognito-identity-provider/dist-es/commands/AdminUpdateDeviceStatusCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AdminUpdateDeviceStatusRequest, AdminUpdateDeviceStatusResponse } from "../models/models_0";
import { deserializeAws_json1_1AdminUpdateDeviceStatusCommand, serializeAws_json1_1AdminUpdateDeviceStatusCommand, } from "../protocols/Aws_json1_1";
var AdminUpdateDeviceStatusCommand = (function (_super) {
    __extends(AdminUpdateDeviceStatusCommand, _super);
    function AdminUpdateDeviceStatusCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AdminUpdateDeviceStatusCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "AdminUpdateDeviceStatusCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AdminUpdateDeviceStatusRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AdminUpdateDeviceStatusResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AdminUpdateDeviceStatusCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AdminUpdateDeviceStatusCommand(input, context);
    };
    AdminUpdateDeviceStatusCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AdminUpdateDeviceStatusCommand(output, context);
    };
    return AdminUpdateDeviceStatusCommand;
}($Command));
export { AdminUpdateDeviceStatusCommand };
