// @aws-sdk/client-cognito-identity-provider/dist-es/commands/AdminListGroupsForUserCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AdminListGroupsForUserRequest, AdminListGroupsForUserResponse } from "../models/models_0";
import { deserializeAws_json1_1AdminListGroupsForUserCommand, serializeAws_json1_1AdminListGroupsForUserCommand, } from "../protocols/Aws_json1_1";
var AdminListGroupsForUserCommand = (function (_super) {
    __extends(AdminListGroupsForUserCommand, _super);
    function AdminListGroupsForUserCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AdminListGroupsForUserCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "AdminListGroupsForUserCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AdminListGroupsForUserRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AdminListGroupsForUserResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AdminListGroupsForUserCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AdminListGroupsForUserCommand(input, context);
    };
    AdminListGroupsForUserCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AdminListGroupsForUserCommand(output, context);
    };
    return AdminListGroupsForUserCommand;
}($Command));
export { AdminListGroupsForUserCommand };
