// @aws-sdk/client-cognito-identity-provider/dist-es/commands/AdminRemoveUserFromGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AdminRemoveUserFromGroupRequest } from "../models/models_0";
import { deserializeAws_json1_1AdminRemoveUserFromGroupCommand, serializeAws_json1_1AdminRemoveUserFromGroupCommand, } from "../protocols/Aws_json1_1";
var AdminRemoveUserFromGroupCommand = (function (_super) {
    __extends(AdminRemoveUserFromGroupCommand, _super);
    function AdminRemoveUserFromGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AdminRemoveUserFromGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "AdminRemoveUserFromGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AdminRemoveUserFromGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AdminRemoveUserFromGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AdminRemoveUserFromGroupCommand(input, context);
    };
    AdminRemoveUserFromGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AdminRemoveUserFromGroupCommand(output, context);
    };
    return AdminRemoveUserFromGroupCommand;
}($Command));
export { AdminRemoveUserFromGroupCommand };
