// @aws-sdk/client-cognito-identity-provider/dist-es/commands/AdminUserGlobalSignOutCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AdminUserGlobalSignOutRequest, AdminUserGlobalSignOutResponse } from "../models/models_0";
import { deserializeAws_json1_1AdminUserGlobalSignOutCommand, serializeAws_json1_1AdminUserGlobalSignOutCommand, } from "../protocols/Aws_json1_1";
var AdminUserGlobalSignOutCommand = (function (_super) {
    __extends(AdminUserGlobalSignOutCommand, _super);
    function AdminUserGlobalSignOutCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AdminUserGlobalSignOutCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "AdminUserGlobalSignOutCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AdminUserGlobalSignOutRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AdminUserGlobalSignOutResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AdminUserGlobalSignOutCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AdminUserGlobalSignOutCommand(input, context);
    };
    AdminUserGlobalSignOutCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AdminUserGlobalSignOutCommand(output, context);
    };
    return AdminUserGlobalSignOutCommand;
}($Command));
export { AdminUserGlobalSignOutCommand };
