// @aws-sdk/client-cognito-identity-provider/dist-es/commands/AdminListDevicesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AdminListDevicesRequest, AdminListDevicesResponse } from "../models/models_0";
import { deserializeAws_json1_1AdminListDevicesCommand, serializeAws_json1_1AdminListDevicesCommand, } from "../protocols/Aws_json1_1";
var AdminListDevicesCommand = (function (_super) {
    __extends(AdminListDevicesCommand, _super);
    function AdminListDevicesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AdminListDevicesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "AdminListDevicesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AdminListDevicesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AdminListDevicesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AdminListDevicesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AdminListDevicesCommand(input, context);
    };
    AdminListDevicesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AdminListDevicesCommand(output, context);
    };
    return AdminListDevicesCommand;
}($Command));
export { AdminListDevicesCommand };
