// @aws-sdk/client-cognito-identity-provider/dist-es/commands/AdminGetUserCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AdminGetUserRequest, AdminGetUserResponse } from "../models/models_0";
import { deserializeAws_json1_1AdminGetUserCommand, serializeAws_json1_1AdminGetUserCommand, } from "../protocols/Aws_json1_1";
var AdminGetUserCommand = (function (_super) {
    __extends(AdminGetUserCommand, _super);
    function AdminGetUserCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AdminGetUserCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "AdminGetUserCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AdminGetUserRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AdminGetUserResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AdminGetUserCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AdminGetUserCommand(input, context);
    };
    AdminGetUserCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AdminGetUserCommand(output, context);
    };
    return AdminGetUserCommand;
}($Command));
export { AdminGetUserCommand };
