// @aws-sdk/client-cognito-identity-provider/dist-es/commands/AdminListUserAuthEventsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AdminListUserAuthEventsRequest, AdminListUserAuthEventsResponse } from "../models/models_0";
import { deserializeAws_json1_1AdminListUserAuthEventsCommand, serializeAws_json1_1AdminListUserAuthEventsCommand, } from "../protocols/Aws_json1_1";
var AdminListUserAuthEventsCommand = (function (_super) {
    __extends(AdminListUserAuthEventsCommand, _super);
    function AdminListUserAuthEventsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AdminListUserAuthEventsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "AdminListUserAuthEventsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AdminListUserAuthEventsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AdminListUserAuthEventsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AdminListUserAuthEventsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AdminListUserAuthEventsCommand(input, context);
    };
    AdminListUserAuthEventsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AdminListUserAuthEventsCommand(output, context);
    };
    return AdminListUserAuthEventsCommand;
}($Command));
export { AdminListUserAuthEventsCommand };
