// @aws-sdk/client-cognito-identity-provider/dist-es/commands/AdminCreateUserCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AdminCreateUserRequest, AdminCreateUserResponse } from "../models/models_0";
import { deserializeAws_json1_1AdminCreateUserCommand, serializeAws_json1_1AdminCreateUserCommand, } from "../protocols/Aws_json1_1";
var AdminCreateUserCommand = (function (_super) {
    __extends(AdminCreateUserCommand, _super);
    function AdminCreateUserCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AdminCreateUserCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "AdminCreateUserCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AdminCreateUserRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AdminCreateUserResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AdminCreateUserCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AdminCreateUserCommand(input, context);
    };
    AdminCreateUserCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AdminCreateUserCommand(output, context);
    };
    return AdminCreateUserCommand;
}($Command));
export { AdminCreateUserCommand };
