// @aws-sdk/client-cognito-identity-provider/dist-es/commands/AssociateSoftwareTokenCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AssociateSoftwareTokenRequest, AssociateSoftwareTokenResponse } from "../models/models_0";
import { deserializeAws_json1_1AssociateSoftwareTokenCommand, serializeAws_json1_1AssociateSoftwareTokenCommand, } from "../protocols/Aws_json1_1";
var AssociateSoftwareTokenCommand = (function (_super) {
    __extends(AssociateSoftwareTokenCommand, _super);
    function AssociateSoftwareTokenCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AssociateSoftwareTokenCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "AssociateSoftwareTokenCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AssociateSoftwareTokenRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AssociateSoftwareTokenResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AssociateSoftwareTokenCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AssociateSoftwareTokenCommand(input, context);
    };
    AssociateSoftwareTokenCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AssociateSoftwareTokenCommand(output, context);
    };
    return AssociateSoftwareTokenCommand;
}($Command));
export { AssociateSoftwareTokenCommand };
