// @aws-sdk/client-cognito-identity-provider/dist-es/commands/ConfirmForgotPasswordCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ConfirmForgotPasswordRequest, ConfirmForgotPasswordResponse } from "../models/models_0";
import { deserializeAws_json1_1ConfirmForgotPasswordCommand, serializeAws_json1_1ConfirmForgotPasswordCommand, } from "../protocols/Aws_json1_1";
var ConfirmForgotPasswordCommand = (function (_super) {
    __extends(ConfirmForgotPasswordCommand, _super);
    function ConfirmForgotPasswordCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ConfirmForgotPasswordCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "ConfirmForgotPasswordCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ConfirmForgotPasswordRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ConfirmForgotPasswordResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ConfirmForgotPasswordCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ConfirmForgotPasswordCommand(input, context);
    };
    ConfirmForgotPasswordCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ConfirmForgotPasswordCommand(output, context);
    };
    return ConfirmForgotPasswordCommand;
}($Command));
export { ConfirmForgotPasswordCommand };
