// @aws-sdk/client-cognito-identity-provider/dist-es/commands/ConfirmSignUpCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ConfirmSignUpRequest, ConfirmSignUpResponse } from "../models/models_0";
import { deserializeAws_json1_1ConfirmSignUpCommand, serializeAws_json1_1ConfirmSignUpCommand, } from "../protocols/Aws_json1_1";
var ConfirmSignUpCommand = (function (_super) {
    __extends(ConfirmSignUpCommand, _super);
    function ConfirmSignUpCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ConfirmSignUpCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "ConfirmSignUpCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ConfirmSignUpRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ConfirmSignUpResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ConfirmSignUpCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ConfirmSignUpCommand(input, context);
    };
    ConfirmSignUpCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ConfirmSignUpCommand(output, context);
    };
    return ConfirmSignUpCommand;
}($Command));
export { ConfirmSignUpCommand };
