// @aws-sdk/client-cognito-identity-provider/dist-es/commands/CreateResourceServerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateResourceServerRequest, CreateResourceServerResponse } from "../models/models_0";
import { deserializeAws_json1_1CreateResourceServerCommand, serializeAws_json1_1CreateResourceServerCommand, } from "../protocols/Aws_json1_1";
var CreateResourceServerCommand = (function (_super) {
    __extends(CreateResourceServerCommand, _super);
    function CreateResourceServerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateResourceServerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "CreateResourceServerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateResourceServerRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateResourceServerResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateResourceServerCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateResourceServerCommand(input, context);
    };
    CreateResourceServerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateResourceServerCommand(output, context);
    };
    return CreateResourceServerCommand;
}($Command));
export { CreateResourceServerCommand };
