// @aws-sdk/client-cognito-identity-provider/dist-es/commands/DeleteUserPoolDomainCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteUserPoolDomainRequest, DeleteUserPoolDomainResponse } from "../models/models_0";
import { deserializeAws_json1_1DeleteUserPoolDomainCommand, serializeAws_json1_1DeleteUserPoolDomainCommand, } from "../protocols/Aws_json1_1";
var DeleteUserPoolDomainCommand = (function (_super) {
    __extends(DeleteUserPoolDomainCommand, _super);
    function DeleteUserPoolDomainCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteUserPoolDomainCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "DeleteUserPoolDomainCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteUserPoolDomainRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteUserPoolDomainResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteUserPoolDomainCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeleteUserPoolDomainCommand(input, context);
    };
    DeleteUserPoolDomainCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeleteUserPoolDomainCommand(output, context);
    };
    return DeleteUserPoolDomainCommand;
}($Command));
export { DeleteUserPoolDomainCommand };
