// @aws-sdk/client-cognito-identity-provider/dist-es/commands/DescribeResourceServerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeResourceServerRequest, DescribeResourceServerResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeResourceServerCommand, serializeAws_json1_1DescribeResourceServerCommand, } from "../protocols/Aws_json1_1";
var DescribeResourceServerCommand = (function (_super) {
    __extends(DescribeResourceServerCommand, _super);
    function DescribeResourceServerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeResourceServerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "DescribeResourceServerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeResourceServerRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeResourceServerResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeResourceServerCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeResourceServerCommand(input, context);
    };
    DescribeResourceServerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeResourceServerCommand(output, context);
    };
    return DescribeResourceServerCommand;
}($Command));
export { DescribeResourceServerCommand };
