// @aws-sdk/client-cognito-identity-provider/dist-es/commands/GetCSVHeaderCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetCSVHeaderRequest, GetCSVHeaderResponse } from "../models/models_0";
import { deserializeAws_json1_1GetCSVHeaderCommand, serializeAws_json1_1GetCSVHeaderCommand, } from "../protocols/Aws_json1_1";
var GetCSVHeaderCommand = (function (_super) {
    __extends(GetCSVHeaderCommand, _super);
    function GetCSVHeaderCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetCSVHeaderCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "GetCSVHeaderCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetCSVHeaderRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetCSVHeaderResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetCSVHeaderCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetCSVHeaderCommand(input, context);
    };
    GetCSVHeaderCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetCSVHeaderCommand(output, context);
    };
    return GetCSVHeaderCommand;
}($Command));
export { GetCSVHeaderCommand };
