// @aws-sdk/client-cognito-identity-provider/dist-es/commands/GetDeviceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetDeviceRequest, GetDeviceResponse } from "../models/models_0";
import { deserializeAws_json1_1GetDeviceCommand, serializeAws_json1_1GetDeviceCommand } from "../protocols/Aws_json1_1";
var GetDeviceCommand = (function (_super) {
    __extends(GetDeviceCommand, _super);
    function GetDeviceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetDeviceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "GetDeviceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetDeviceRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetDeviceResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetDeviceCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetDeviceCommand(input, context);
    };
    GetDeviceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetDeviceCommand(output, context);
    };
    return GetDeviceCommand;
}($Command));
export { GetDeviceCommand };
