// @aws-sdk/client-cognito-identity-provider/dist-es/commands/GetGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetGroupRequest, GetGroupResponse } from "../models/models_0";
import { deserializeAws_json1_1GetGroupCommand, serializeAws_json1_1GetGroupCommand } from "../protocols/Aws_json1_1";
var GetGroupCommand = (function (_super) {
    __extends(GetGroupCommand, _super);
    function GetGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "GetGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetGroupCommand(input, context);
    };
    GetGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetGroupCommand(output, context);
    };
    return GetGroupCommand;
}($Command));
export { GetGroupCommand };
