// @aws-sdk/client-cognito-identity-provider/dist-es/commands/GetIdentityProviderByIdentifierCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetIdentityProviderByIdentifierRequest, GetIdentityProviderByIdentifierResponse } from "../models/models_0";
import { deserializeAws_json1_1GetIdentityProviderByIdentifierCommand, serializeAws_json1_1GetIdentityProviderByIdentifierCommand, } from "../protocols/Aws_json1_1";
var GetIdentityProviderByIdentifierCommand = (function (_super) {
    __extends(GetIdentityProviderByIdentifierCommand, _super);
    function GetIdentityProviderByIdentifierCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetIdentityProviderByIdentifierCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "GetIdentityProviderByIdentifierCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetIdentityProviderByIdentifierRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetIdentityProviderByIdentifierResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetIdentityProviderByIdentifierCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetIdentityProviderByIdentifierCommand(input, context);
    };
    GetIdentityProviderByIdentifierCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetIdentityProviderByIdentifierCommand(output, context);
    };
    return GetIdentityProviderByIdentifierCommand;
}($Command));
export { GetIdentityProviderByIdentifierCommand };
