// @aws-sdk/client-cognito-identity-provider/dist-es/commands/GetUserAttributeVerificationCodeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetUserAttributeVerificationCodeRequest, GetUserAttributeVerificationCodeResponse } from "../models/models_0";
import { deserializeAws_json1_1GetUserAttributeVerificationCodeCommand, serializeAws_json1_1GetUserAttributeVerificationCodeCommand, } from "../protocols/Aws_json1_1";
var GetUserAttributeVerificationCodeCommand = (function (_super) {
    __extends(GetUserAttributeVerificationCodeCommand, _super);
    function GetUserAttributeVerificationCodeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetUserAttributeVerificationCodeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "GetUserAttributeVerificationCodeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetUserAttributeVerificationCodeRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetUserAttributeVerificationCodeResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetUserAttributeVerificationCodeCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetUserAttributeVerificationCodeCommand(input, context);
    };
    GetUserAttributeVerificationCodeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetUserAttributeVerificationCodeCommand(output, context);
    };
    return GetUserAttributeVerificationCodeCommand;
}($Command));
export { GetUserAttributeVerificationCodeCommand };
