// @aws-sdk/client-cognito-identity-provider/dist-es/commands/GetUserCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetUserRequest, GetUserResponse } from "../models/models_0";
import { deserializeAws_json1_1GetUserCommand, serializeAws_json1_1GetUserCommand } from "../protocols/Aws_json1_1";
var GetUserCommand = (function (_super) {
    __extends(GetUserCommand, _super);
    function GetUserCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetUserCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "GetUserCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetUserRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetUserResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetUserCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetUserCommand(input, context);
    };
    GetUserCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetUserCommand(output, context);
    };
    return GetUserCommand;
}($Command));
export { GetUserCommand };
