// @aws-sdk/client-cognito-identity-provider/dist-es/commands/InitiateAuthCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { InitiateAuthRequest, InitiateAuthResponse } from "../models/models_0";
import { deserializeAws_json1_1InitiateAuthCommand, serializeAws_json1_1InitiateAuthCommand, } from "../protocols/Aws_json1_1";
var InitiateAuthCommand = (function (_super) {
    __extends(InitiateAuthCommand, _super);
    function InitiateAuthCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    InitiateAuthCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "InitiateAuthCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: InitiateAuthRequest.filterSensitiveLog,
            outputFilterSensitiveLog: InitiateAuthResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    InitiateAuthCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1InitiateAuthCommand(input, context);
    };
    InitiateAuthCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1InitiateAuthCommand(output, context);
    };
    return InitiateAuthCommand;
}($Command));
export { InitiateAuthCommand };
