// @aws-sdk/client-cognito-identity-provider/dist-es/commands/ListUsersInGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListUsersInGroupRequest, ListUsersInGroupResponse } from "../models/models_0";
import { deserializeAws_json1_1ListUsersInGroupCommand, serializeAws_json1_1ListUsersInGroupCommand, } from "../protocols/Aws_json1_1";
var ListUsersInGroupCommand = (function (_super) {
    __extends(ListUsersInGroupCommand, _super);
    function ListUsersInGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListUsersInGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "ListUsersInGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListUsersInGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListUsersInGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListUsersInGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListUsersInGroupCommand(input, context);
    };
    ListUsersInGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListUsersInGroupCommand(output, context);
    };
    return ListUsersInGroupCommand;
}($Command));
export { ListUsersInGroupCommand };
