// @aws-sdk/client-cognito-identity-provider/dist-es/commands/ResendConfirmationCodeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ResendConfirmationCodeRequest, ResendConfirmationCodeResponse } from "../models/models_0";
import { deserializeAws_json1_1ResendConfirmationCodeCommand, serializeAws_json1_1ResendConfirmationCodeCommand, } from "../protocols/Aws_json1_1";
var ResendConfirmationCodeCommand = (function (_super) {
    __extends(ResendConfirmationCodeCommand, _super);
    function ResendConfirmationCodeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ResendConfirmationCodeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "ResendConfirmationCodeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ResendConfirmationCodeRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ResendConfirmationCodeResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ResendConfirmationCodeCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ResendConfirmationCodeCommand(input, context);
    };
    ResendConfirmationCodeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ResendConfirmationCodeCommand(output, context);
    };
    return ResendConfirmationCodeCommand;
}($Command));
export { ResendConfirmationCodeCommand };
