// @aws-sdk/client-cognito-identity-provider/dist-es/commands/SignUpCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SignUpRequest, SignUpResponse } from "../models/models_0";
import { deserializeAws_json1_1SignUpCommand, serializeAws_json1_1SignUpCommand } from "../protocols/Aws_json1_1";
var SignUpCommand = (function (_super) {
    __extends(SignUpCommand, _super);
    function SignUpCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SignUpCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "SignUpCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SignUpRequest.filterSensitiveLog,
            outputFilterSensitiveLog: SignUpResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SignUpCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1SignUpCommand(input, context);
    };
    SignUpCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1SignUpCommand(output, context);
    };
    return SignUpCommand;
}($Command));
export { SignUpCommand };
