// @aws-sdk/client-cognito-identity-provider/dist-es/commands/UpdateIdentityProviderCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateIdentityProviderRequest, UpdateIdentityProviderResponse } from "../models/models_1";
import { deserializeAws_json1_1UpdateIdentityProviderCommand, serializeAws_json1_1UpdateIdentityProviderCommand, } from "../protocols/Aws_json1_1";
var UpdateIdentityProviderCommand = (function (_super) {
    __extends(UpdateIdentityProviderCommand, _super);
    function UpdateIdentityProviderCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateIdentityProviderCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "UpdateIdentityProviderCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateIdentityProviderRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateIdentityProviderResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateIdentityProviderCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateIdentityProviderCommand(input, context);
    };
    UpdateIdentityProviderCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateIdentityProviderCommand(output, context);
    };
    return UpdateIdentityProviderCommand;
}($Command));
export { UpdateIdentityProviderCommand };
