// @aws-sdk/client-cognito-identity-provider/dist-es/commands/UpdateResourceServerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateResourceServerRequest, UpdateResourceServerResponse } from "../models/models_1";
import { deserializeAws_json1_1UpdateResourceServerCommand, serializeAws_json1_1UpdateResourceServerCommand, } from "../protocols/Aws_json1_1";
var UpdateResourceServerCommand = (function (_super) {
    __extends(UpdateResourceServerCommand, _super);
    function UpdateResourceServerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateResourceServerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "UpdateResourceServerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateResourceServerRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateResourceServerResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateResourceServerCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateResourceServerCommand(input, context);
    };
    UpdateResourceServerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateResourceServerCommand(output, context);
    };
    return UpdateResourceServerCommand;
}($Command));
export { UpdateResourceServerCommand };
