// @aws-sdk/client-cognito-identity-provider/dist-es/commands/UpdateDeviceStatusCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateDeviceStatusRequest, UpdateDeviceStatusResponse } from "../models/models_1";
import { deserializeAws_json1_1UpdateDeviceStatusCommand, serializeAws_json1_1UpdateDeviceStatusCommand, } from "../protocols/Aws_json1_1";
var UpdateDeviceStatusCommand = (function (_super) {
    __extends(UpdateDeviceStatusCommand, _super);
    function UpdateDeviceStatusCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateDeviceStatusCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "UpdateDeviceStatusCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateDeviceStatusRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateDeviceStatusResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateDeviceStatusCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateDeviceStatusCommand(input, context);
    };
    UpdateDeviceStatusCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateDeviceStatusCommand(output, context);
    };
    return UpdateDeviceStatusCommand;
}($Command));
export { UpdateDeviceStatusCommand };
