// @aws-sdk/client-cognito-identity-provider/dist-es/commands/UpdateUserPoolDomainCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateUserPoolDomainRequest, UpdateUserPoolDomainResponse } from "../models/models_1";
import { deserializeAws_json1_1UpdateUserPoolDomainCommand, serializeAws_json1_1UpdateUserPoolDomainCommand, } from "../protocols/Aws_json1_1";
var UpdateUserPoolDomainCommand = (function (_super) {
    __extends(UpdateUserPoolDomainCommand, _super);
    function UpdateUserPoolDomainCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateUserPoolDomainCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "UpdateUserPoolDomainCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateUserPoolDomainRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateUserPoolDomainResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateUserPoolDomainCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateUserPoolDomainCommand(input, context);
    };
    UpdateUserPoolDomainCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateUserPoolDomainCommand(output, context);
    };
    return UpdateUserPoolDomainCommand;
}($Command));
export { UpdateUserPoolDomainCommand };
