// @aws-sdk/client-cognito-identity-provider/dist-es/commands/VerifySoftwareTokenCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getAwsAuthPlugin } from "@aws-sdk/middleware-signing";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { VerifySoftwareTokenRequest, VerifySoftwareTokenResponse } from "../models/models_1";
import { deserializeAws_json1_1VerifySoftwareTokenCommand, serializeAws_json1_1VerifySoftwareTokenCommand, } from "../protocols/Aws_json1_1";
var VerifySoftwareTokenCommand = (function (_super) {
    __extends(VerifySoftwareTokenCommand, _super);
    function VerifySoftwareTokenCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    VerifySoftwareTokenCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getAwsAuthPlugin(configuration));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CognitoIdentityProviderClient";
        var commandName = "VerifySoftwareTokenCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: VerifySoftwareTokenRequest.filterSensitiveLog,
            outputFilterSensitiveLog: VerifySoftwareTokenResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    VerifySoftwareTokenCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1VerifySoftwareTokenCommand(input, context);
    };
    VerifySoftwareTokenCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1VerifySoftwareTokenCommand(output, context);
    };
    return VerifySoftwareTokenCommand;
}($Command));
export { VerifySoftwareTokenCommand };
